order-book/src/main/java/com/market/banica/order/book/model/ItemMarket.java

```diff
@@ -37,22 +54,116 @@ public void removeUntrackedItem(String itemName) {
         allItems.remove(itemName);
     }
 
+    public void updateItem(Aurora.AuroraResponse response) {
+        if (response.getMessage().is(TickResponse.class)) {
+            TickResponse tickResponse;
+
+            try {
+                tickResponse = response.getMessage().unpack(TickResponse.class);
+            } catch (InvalidProtocolBufferException e) {
+                throw new IncorrectResponseException("Response is not correct!");
+            }
+
+            Item item = new Item();
+            item.setPrice(tickResponse.getPrice());
+            item.setQuantity(tickResponse.getQuantity());
+            item.setOrigin(tickResponse.getOrigin());
+
+            productsQuantity.merge(tickResponse.getGoodName(), tickResponse.getQuantity(), Long::sum);
+
+            Set<Item> itemSet = allItems.get(tickResponse.getGoodName());
+            if (itemSet != null) {
+                if (itemSet.contains(item)) {
```

Comment:
could we merge the 2 if statements as itemSet != null && itemSet.contains(item)

Reply:
Fixed